### PR TITLE
Changes to add ready callback. Used to keep track of startup metrics

### PR DESCRIPTION
**Pull request by johnmuth81**
This is the KM needed for the new kontain-faas backend.

### CODE REVIEW

File: km/km_filesys.c [hunk 1]
```diff
@@ -3795,6 +3800,38 @@ static int km_fs_recover_open_socket(char* ptr, size_t length)
 
    struct sockaddr* sa = (struct sockaddr*)(ptr + sizeof(km_nt_socket_t));
    km_fs_recover_socket(nt_sock, sa, nt_sock->addrlen);
+#if 0
```
Comment (gnode1) [inline km/km_filesys.c:3803]: Merge cruft? This logic moved

File: km/km_main.c [hunk 2]
```diff
@@ -102,7 +102,10 @@ static inline void usage()
 "\t--virt-device=<file-name>  (-Ffile) - Use provided file-name for virtualization device\n"
 "\t--input-data=<file-name>            - File with data for HC_snapshot_getdata\n"
 "\t--output-data=<file-name>           - File with data from HC_snapshot_putdata\n"
-"\t--mgtpipe <path>                    - Name for management pipe.\n");
+"\t--no-log-redirect                   - Do not redirect logs.\n"
```
Comment (gnode1) [inline km/km_main.c:105]: Why/when do we need this?

File: km/km_management.c [hunk 3]
```diff
@@ -184,3 +189,102 @@ void km_mgt_init(char* path)
...
    return;
 }
+
+static int cb_domain = AF_INET;
+static int cb_type = SOCK_DGRAM;
+static struct sockaddr* cb_sockaddr = NULL;
+static int cb_sockaddr_len = 0;
+
+int
+km_build_ipaddress(char *host, char* port)
```
Comment (gnode1) [inline km/km_management.c:199]: static?

File: km/km_management.c [hunk 4]
```diff
@@ -184,3 +189,102 @@ void km_mgt_init(char* path)
...
+   int sock = socket(cb_domain, cb_type, 0);
+   if (sock < 0) {
+      km_warn("callback socket create");
+   }
+
+   if (cb_domain == AF_INET && cb_type == SOCK_DGRAM) {
+      if (sendto(sock, &msg, sizeof(msg), 0, cb_sockaddr, cb_sockaddr_len) < 0) {
+         km_warnx("callback udp sendto failed");
+      }
+   }
```
Comment (paulpopelka) [inline km/km_management.c:278]: close sock here?

File: km/km_management.c [hunk 5]
```diff
@@ -184,3 +189,102 @@ void km_mgt_init(char* path)
...
+
+   int sock = socket(cb_domain, cb_type, 0);
+   if (sock < 0) {
+      km_warn("callback socket create");
+   }
+
+   if (cb_domain == AF_INET && cb_type == SOCK_DGRAM) {
+      if (sendto(sock, &msg, sizeof(msg), 0, cb_sockaddr, cb_sockaddr_len) < 0) {
+         km_warnx("callback udp sendto failed");
```
Comment (paulpopelka) [inline km/km_management.c:276]: You probably want km_warn() here to get the errno printed.

File: km/km_filesys.c (hunk 1 above)
Comment (johnmuth81) [reply km/km_filesys.c:3821]: fixed.

File: km/km_main.c (hunk 2 above)
Comment (johnmuth81) [reply km/km_main.c:105]: We don't. Removed.

File: km/km_management.c (hunk 3 above)
Comment (johnmuth81) [reply km/km_management.c]: done

File: km/km_management.c (hunk 5 above)
Comment (johnmuth81) [reply km/km_management.c]: done

File: km/km_management.c [hunk 6]
```diff
@@ -184,3 +189,149 @@ void km_mgt_init(char* path)
...
+
+   if (sendto(sock, &msg, sizeof(msg), 0, cb_sockaddr, cb_sockaddr_len) < 0) {
+      km_warn("callback udp sendto failed");
+   }
+   close(sock);
+}
+
+void km_fire_km_started_callback()
+{
+   km_fire_callback(0x304d4b);
```
Comment (paulpopelka) [inline km/km_management.c:322]: Where do these hex values come from?

0x304d4b
0x314d4b
0x324d4b

Comment (johnmuth81) [reply km/km_management.c:322]: "KM0". "KM1". "KM2" in ascii...

**Comment by gnode1** [comment]
obsolete. FaaS server keeps the stats

**Pull request closed without merging**